homeassistant/config_entries.py

```diff
@@ -911,6 +927,48 @@ async def async_step_unignore(self, user_input: Dict[str, Any]) -> Dict[str, Any
         """Rediscover a config entry by it's unique_id."""
         return self.async_abort(reason="not_implemented")
 
+    async def async_step_user(
+        self, user_input: Optional[Dict[str, Any]] = None
+    ) -> Dict[str, Any]:
+        """Handle a flow initiated by the user."""
+        return self.async_abort(reason="not_implemented")
+
+    async def async_handle_discovery_without_unique_id(self) -> None:
+        """Mark this flow discovered, without a unique identifier.
+
+        If a flow initiated by discovery, doesn't have a unique ID, this can
+        be used alternatively. It will ensure only 1 flow is started and only
+        when the handler has no existing config entries.
+
+        It ensures that the discovery can be ignored by the user.
+        """
+        if self.unique_id is not None:
+            return
```

Comment:
Can this happen?

Reply:
I think it makes the implementation easier, as it allows one to call this method without having to worry about it.
I've added the IPP example use-case to the PR description to show why.

Reply:
Nice!